src/components/Messages/PreviewList.tsx

```diff
@@ -27,8 +29,12 @@ const PreviewList: FC = () => {
   const showLoading = loading && (messages.size === 0 || profiles.size === 0);
 
   const sortedProfiles = Array.from(profiles).sort(([, a], [, b]) => {
-    const messageA = messages.get(a.ownedBy.toLowerCase());
-    const messageB = messages.get(b.ownedBy.toLowerCase());
+    const conversationIdA = buildConversationId(currentProfile?.id, a.id);
```

Comment:
You don't actually need to rebuild the `conversationKey` here. The key in the `profiles` map is the conversationKey

Reply:
LOL great catch 🧠 . So you should be able to do

```js
const sortedProfiles = Array.from(profiles).sort(([keyA, a], [keyB, b]) => {
```

Reply:
Thanks